src/Simulation/Core/EntryPointInfo.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Quantum.Simulation.Core
+{
+    /// <summary>
+    /// Base class containing information about a Q# entry point.
+    /// </summary>
+    public class EntryPointInfo<I, O>
+    {
+        public Type Operation { get; }
+        public Type InType => typeof(I);
+        public Type OutType => typeof(O);
+
+        public EntryPointInfo(Type operation) => 
+            this.Operation = operation;
+    }
+
+    /// <summary>
+    /// Base class containing information about an entry point 
+    /// for a Q# executable targeted for a Alfred quantum processor.
+    /// </summary>
+    public class AlfredExecutable<I, O> 
```

Comment:
Regarding the names of the classes, I think it would be more consistent to have AlfredInfo, BrunoInfo and ClementineInfo as names.

The current names are ok with me if you feel strongly about keeping them.

Reply:
AlfredInfo seems slightly misleading, since it doesn't actually contain infos about Alfred. How about renaming EntryPointInfo to Executable, or renaming AlfredExecutable to AlfreadEntryPointInfo etc?

Reply:
Going with AlfredEntryPointInfo etc for now - I'm happy to change it later, when make the pass to adapt these names, since the names are only temporary.